metadata-ingestion/tests/integration/tableau/test_tableau_ingest.py

```diff
@@ -755,30 +790,48 @@ def test_tableau_unsupported_csql(mock_datahub_graph):
     config.lineage_overrides = TableauLineageOverrides(
         database_override_map={"production database": "prod"}
     )
-    source = TableauSource(config=config, ctx=context)
-    lineage = source._create_lineage_from_unsupported_csql(
-        csql_urn="urn:li:dataset:(urn:li:dataPlatform:tableau,09988088-05ad-173c-a2f1-f33ba3a13d1a,PROD)",
-        csql={
-            "query": "SELECT user_id, source, user_source FROM (SELECT *, ROW_NUMBER() OVER (partition BY user_id ORDER BY __partition_day DESC) AS rank_ FROM invent_dw.UserDetail ) source_user WHERE rank_ = 1",
-            "isUnsupportedCustomSql": "true",
-            "database": {"name": "production database", "connectionType": "bigquery"},
-        },
-    )
 
-    mcp = cast(MetadataChangeProposalClass, next(iter(lineage)).metadata)
+    with mock.patch(
+        "datahub.ingestion.source.tableau.sqlglot_lineage"
+    ) as sqlglot_lineage:
 
-    assert mcp.aspect == UpstreamLineage(
-        upstreams=[
-            UpstreamClass(
-                dataset="urn:li:dataset:(urn:li:dataPlatform:bigquery,prod.invent_dw.userdetail,PROD)",
-                type=DatasetLineageType.TRANSFORMED,
-            )
-        ]
-    )
-    assert (
-        mcp.entityUrn
-        == "urn:li:dataset:(urn:li:dataPlatform:tableau,09988088-05ad-173c-a2f1-f33ba3a13d1a,PROD)"
-    )
+        sqlglot_lineage.return_value = SqlParsingResult(  # type:ignore
+            in_tables=[
+                "urn:li:dataset:(urn:li:dataPlatform:bigquery,invent_dw.userdetail,PROD)"
+            ],
+            out_tables=[],
+            column_lineage=None,
+        )
+
+        source = TableauSource(config=config, ctx=context)
+
+        lineage = source._create_lineage_from_unsupported_csql(
+            csql_urn="urn:li:dataset:(urn:li:dataPlatform:tableau,09988088-05ad-173c-a2f1-f33ba3a13d1a,PROD)",
+            csql={
+                "query": "SELECT user_id, source, user_source FROM (SELECT *, ROW_NUMBER() OVER (partition BY user_id ORDER BY __partition_day DESC) AS rank_ FROM invent_dw.UserDetail ) source_user WHERE rank_ = 1",
+                "isUnsupportedCustomSql": "true",
+                "database": {
+                    "name": "production database",
```

Comment:
let's change this to be `my-bigquery-project`

Given that it's set, shouldn't this be used as upstream_db, and eventually passed as default_db to sqlglot_lineage?

Reply:
done. It is getting used as default_db 
code link: https://github.com/datahub-project/datahub/pull/8466/files#diff-f618f3bf4fb8b6cd799b3375ab22051094345681942a27895f8a7d56f3c36951R1606